edward/inferences.py

```diff
@@ -12,67 +12,85 @@
     pass
 
 class Inference:
-    """
-    Base class for inference methods.
-
-    Parameters
-    ----------
-    model : Model
-        probability model p(x, z)
-    data : Data, optional
-        data x
+    """Base class for Edward inference methods.
     """
     def __init__(self, model, data=Data()):
+        """Initialization.
+
+        Calls ``util.get_session()``
+        
+        Parameters
+        ----------
+        model : ed.Model
+            probability model
+        data : ed.Data, optional
+            observed data
+        """
         self.model = model
         self.data = data
         get_session()
 
 class MonteCarlo(Inference):
-    """
-    Base class for Monte Carlo methods.
-
-    Parameters
-    ----------
-    model : Model
-        probability model p(x, z)
-    data : Data, optional
-        data x
+    """Base class for Monte Carlo inference methods.
     """
     def __init__(self, *args, **kwargs):
+        """Initialization.
+
+        Parameters
+        ----------
+        model : ed.Model
+            probability model
+        data : ed.Data, optional
+            observed data
+        """
         Inference.__init__(self, *args, **kwargs)
 
 class VariationalInference(Inference):
-    """
-    Base class for variational inference methods.
-
-    Parameters
-    ----------
-    model : Model
-        probability model p(x, z)
-    variational : Variational
-        variational model q(z; lambda)
-    data : Data, optional
-        data x
+    """Base class for variational inference methods.
     """
     def __init__(self, model, variational, data=Data()):
+        """Initialization.
+
+        Parameters
+        ----------
+        model : ed.Model
+            probability model
+        variational : ed.Variational
+            variational model or distribution
+        data : ed.Data, optional
+            observed data        
+        """
         Inference.__init__(self, model, data)
         self.variational = variational
 
     def run(self, *args, **kwargs):
-        """
-        A simple wrapper to run the inference algorithm.
+        """A simple wrapper to run variational inference.
+
+        1. Initialize via ``initialize``
+        2. Run ``update`` for ``self.n_iter`` iterations 
+        3. While running, ``print_progress``
+        4. Finalize via ``finalize``
+
+        Parameters
+        ----------
+        *args :
+            goes nowhere
```

Comment:
i'm afraid i didn't understand your comment.

more generally: i'm not sure how to document this kind of wrappers of wrappers settings properly.